File: problem1.rb
# If we list all the natural numbers below 10 that are
# multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
# Find the sum of all the multiples of 3 or 5 below 1000.

def is_multiple(n)
    if n % 3 == 0
        return true;
    end
    if n % 5 == 0
        return true;
    else
        return false;
    end
end

num = (1..999).select { |e| is_multiple(e) }
sum = 0
num.each {|e| sum += e}
puts sum

File: clj/src/clj/problem1.clj
(ns clj.problem1)

; If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9.
; The sum of these multiples is 23.
; Find the sum of all the multiples of 3 or 5 below 1000.

(defn multiple-of-3-or-5?
  [num]
  (or (= (rem num 3) 0)
      (= (rem num 5) 0)))

(defn filter-multiple-of-3-or-5
  [collection]
  (filter
    multiple-of-3-or-5?
    collection))

(reduce +
  (filter-multiple-of-3-or-5
    (range 1 1000)))
